qa/rpc-tests/wallet_listnotes.py

```diff
@@ -0,0 +1,168 @@
+#!/usr/bin/env python2
+# Copyright (c) 2018 The Zcash developers
+# Distributed under the MIT software license, see the accompanying
+# file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+from test_framework.test_framework import BitcoinTestFramework
+from test_framework.util import assert_equal, start_nodes, wait_and_assert_operationid_status
+
+from decimal import Decimal
+
+# Test wallet z_listunspent behaviour across network upgrades
+class WalletListNotes(BitcoinTestFramework):
+
+    def setup_nodes(self):
+        return start_nodes(4, self.options.tmpdir, [[
+            '-nuparams=5ba81b19:202', # Overwinter
+            '-nuparams=76b809bb:204', # Sapling
+        ]] * 4)
+
+    def run_test(self):
+        # Current height = 200 -> Sprout
+        assert_equal(200, self.nodes[0].getblockcount())
+        sproutzaddr = self.nodes[0].z_getnewaddress('sprout')
+
+        # test that we can create a sapling zaddr before sapling activates
+        saplingzaddr = self.nodes[0].z_getnewaddress('sapling')
+
+        # we've got lots of coinbase (taddr) but no shielded funds yet
+        assert_equal(0, Decimal(self.nodes[0].z_gettotalbalance()['private']))
+        
+        # Set current height to 201 -> Sprout
+        self.nodes[0].generate(1)
+        self.sync_all()
+        assert_equal(201, self.nodes[0].getblockcount())
+
+        mining_addr = self.nodes[0].listunspent()[0]['address']
+
+        # Shield coinbase funds (must be a multiple of 10, no change allowed pre-sapling)
+        receive_amount_10 = Decimal('10.0') - Decimal('0.0001')
+        recipients = [{"address":sproutzaddr, "amount":receive_amount_10}]
+        myopid = self.nodes[0].z_sendmany(mining_addr, recipients)
+        txid_1 = wait_and_assert_operationid_status(self.nodes[0], myopid)
+        self.sync_all()
+        
+        # No funds (with (default) one or more confirmations) in sproutzaddr yet
+        assert_equal(0, len(self.nodes[0].z_listunspent()))
+        assert_equal(0, len(self.nodes[0].z_listunspent(1)))
+        
+        # no private balance because no confirmations yet
+        assert_equal(0, Decimal(self.nodes[0].z_gettotalbalance()['private']))
+        
+        # list private unspent, this time allowing 0 confirmations
+        unspent_cb = self.nodes[0].z_listunspent(0)
+        assert_equal(1, len(unspent_cb))
+        assert_equal(False,             unspent_cb[0]['change'])
+        assert_equal(txid_1,            unspent_cb[0]['txid'])
+        assert_equal(True,              unspent_cb[0]['spendable'])
```

Comment:
Thanks, didn't know about that. I think in this case it's more consistent to keep it as-is because then all the assertions in this group have the same format: compare an expected value with an actual value. Less thinking for the reader.